tests/EStimLibrary.UnitTests/Core/HardwareInterfaces/LeadTests.cs

```diff
@@ -0,0 +1,618 @@
+using EStimLibrary.Core;
+using EStimLibrary.Core.HardwareInterfaces;
+using Xunit;
+using System.Collections.Generic;
+
+namespace EStimLibrary.UnitTests.Core.HardwareInterfaces
+{
+    public class LeadTests
+    {
+        /// <summary>
+        /// Test the parameterized constructor with different data values.
+        /// </summary>
+        [Theory]
+        [MemberData(nameof(ConstructorTestData))]
+        public void ConstructorShouldInitSortedSetsAndDirection(SortedSet<int> contactSet,
+            SortedSet<int> outputSet, Constants.CurrentDirection currentDirection)
+        {
+            // Act
+            var lead = new Lead(contactSet, outputSet, currentDirection);
+
+            // Assert
+            Assert.Equal(contactSet, lead.ContactSet);
+            Assert.Equal(outputSet, lead.OutputSet);
+            Assert.Equal(currentDirection, lead.CurrentDirection);
+        }
+
+        public static IEnumerable<object[]> ConstructorTestData()
+        {
+            return new List<object[]>
+            {
+                new object[]
```

Comment:
Would also just add brief in-line comments for each of your test data inputs to note what you're testing with it, i.e., what's different about the test case to warrant you making it its own test case.